src/snowflake/connector/connection.py

```diff
@@ -329,183 +331,184 @@ def _ocsp_mode(self) -> OCSPMode:
             return OCSPMode.FAIL_CLOSED
 
     @property
-    def session_id(self):
+    def session_id(self) -> int:
         return self._session_id
 
     @property
-    def user(self):
+    def user(self) -> str:
         return self._user
 
     @property
-    def host(self):
+    def host(self) -> str:
         return self._host
 
     @property
-    def port(self):
+    def port(self) -> int | str:  # TODO: shouldn't be a string
```

Comment:
if this shouldn't be a string, then our type probably should not hint it could be a string?
otherwise users could be dependent on the string behavior, and when one day we go with int only, users think we're breaking them..

Reply:
This was just my thought. When I saw this auto generated type-hint. If it's a unit test that calls this with a string and it isn't documented then we can update the type-hint.